src/main.c

```diff
@@ -105,7 +107,8 @@ void loop(struct l3ctx *ctx) {
 					intercom_handle_in(&ctx->intercom_ctx, events[i].data.fd);
 			} else if (ctx->taskqueue_ctx.fd == events[i].data.fd) {
 				taskqueue_run(&ctx->taskqueue_ctx);
-			} else if (ctx->wifistations_ctx.fd == events[i].data.fd) {
+			} else if (ctx->wifistations_ctx.fd
+				  && ctx->wifistations_ctx.fd == events[i].data.fd) {
```

Comment:
This change is not necessary, `events[i].data.fd` will never match `ctx->wifistations_ctx.fd` if `ctx->wifistations_ctx.fd` is not a valid file descriptor.